go/array_string/min_window/solution.go

```diff
@@ -0,0 +1,101 @@
+// package main has the solution for the minimum window
+// algorithm implementation using the sliding window technique.
+package main
+
+import (
+	"fmt"
+	"math"
+	"strings"
+)
+
+func minWindow(mainStr, subStr string) string {
+	// validation
+	if len(mainStr) == 0 || len(subStr) == 0 || len(subStr) > len(mainStr) {
+		return ""
+	}
+	// using the sliding window technique.
+	// create hashmap for required and desired character.
+	required := make(map[string]int)
+	desired := make(map[string]int)
```

Comment:
What do you mean by "required" and "desired"? Is required the total number from the substring? I'm not sure what desired is because it seems more like it's what you've found in the current window

Reply:
You're right, `desired` is what I found in the current window, taking count, while `required` holds the character in the substring and the no of occurrences (in case of repeated characters). I use it to compare what I have in the `desired` hashmap